biorbd_optim/misc/optimal_control_program.py

```diff
@@ -283,38 +256,11 @@ def __init__(
         self.__define_variable_time(initial_time_guess, time_min, time_max)
 
         # Prepare path constraints and dynamics of the program
-        self.__add_to_nlp("X_bounds", X_bounds, False)
-        self.__add_to_nlp("U_bounds", U_bounds, False)
         self.__add_to_nlp("dynamics_type", dynamics_type, False)
         self.__add_to_nlp("ode_solver", ode_solver, True)
         self.__add_to_nlp("control_type", control_type, True)
-        for i in range(self.nb_phases):
-            self.__initialize_nlp(self.nlp[i])
-            Problem.initialize(self, self.nlp[i])
-        for i in range(self.nb_phases):
-            self.nlp[i].X_bounds.check_and_adjust_dimensions(self.nlp[i].nx, self.nlp[i].ns)
-            if self.nlp[i].control_type == ControlType.CONSTANT:
-                self.nlp[i].U_bounds.check_and_adjust_dimensions(self.nlp[i].nu, self.nlp[i].ns - 1)
-            elif self.nlp[i].control_type == ControlType.LINEAR_CONTINUOUS:
-                self.nlp[i].U_bounds.check_and_adjust_dimensions(self.nlp[i].nu, self.nlp[i].ns)
-            else:
-                raise NotImplementedError(f"Plotting {self.nlp[i]['control_type']} is not implemented yet")
 
-        # Prepare initial guesses
-        self.__add_to_nlp("X_init", X_init, False)
-        self.__add_to_nlp("U_init", U_init, False)
-        for i in range(self.nb_phases):
-            self.nlp[i].X_init.check_and_adjust_dimensions(self.nlp[i].nx, self.nlp[i].ns)
-            if self.nlp[i].control_type == ControlType.CONSTANT:
-                self.nlp[i].U_init.check_and_adjust_dimensions(self.nlp[i].nu, self.nlp[i].ns - 1)
-            elif self.nlp[i].control_type == ControlType.LINEAR_CONTINUOUS:
-                self.nlp[i].U_init.check_and_adjust_dimensions(self.nlp[i].nu, self.nlp[i].ns)
-            else:
-                raise NotImplementedError(f"Plotting {self.nlp[i]['control_type']} is not implemented yet")
-
-        # Variables and constraint for the optimization program
-        for i in range(self.nb_phases):
-            self.__define_multiple_shooting_nodes_per_phase(self.nlp[i], i)
```

Comment:
I think it will be necessary somehow to call this function anyway, but not set what is not ready. For instance, if ones provides initial guesses but don't provide bounds, this function should set the initial guesses and V (so V and V_init). If one provides nothing, then it would only set the variable V (but not V_init nor V_bounds)

Reply:
That is because ACADOS does not need initial guess, while Ipopt does. So we should have a flag that remember what is set and make the `solve` fail or succeed depending on what is set